Client/src/Unified/Services/Submitter/Service.cs

```diff
@@ -139,34 +141,91 @@ public class Service : AbstractClientService, ISubmitterService
 
-                                var taskIds =
-                                  SessionService.SubmitTasksWithDependencies(blockRequestList.Select(x => new Tuple<string, byte[], IList<string>>(x.ResultId.ToString(),
-                                                                                                                                                   x.Payload!
-                                                                                                                                                    .Serialize(),
-                                                                                                                                                   null)));
-                                var taskIdsResultIds = SessionService.GetResultIds(taskIds);
 
-                                foreach (var pairTaskIdResultId in taskIdsResultIds)
-                                {
-                                  var blockRequest = blockRequestList.FirstOrDefault(x => x.ResultId.ToString() == pairTaskIdResultId.ResultIds.First());
-                                  if (blockRequest == null)
+                                  for (var retry = 0; retry < MaxRetries; retry++)
                                   {
-                                    throw new InvalidOperationException($"Cannot find BlockRequest with result id {pairTaskIdResultId.TaskId}");
+                                    //Generate resultId
+                                    foreach (var x in blockRequestList)
+                                    {
+                                      x.ResultId = Guid.NewGuid();
+                                    }
+
+                                    try
+                                    {
+                                      var taskIds =
+                                        SessionService.SubmitTasksWithDependencies(blockRequestList.Select(x => new
+                                                                                                             Tuple<string, byte[], IList<string>>(x.ResultId.ToString(),
+                                                                                                                                                  x.Payload!.Serialize(),
+                                                                                                                                                  null)),
+                                                                                   1);
+
+                                      var ids            = taskIds.ToList();
+                                      var mapTaskResults = SessionService.GetResultIds(ids);
+                                      var taskIdsResultIds = mapTaskResults.ToDictionary(result => result.ResultIds.Single(),
+                                                                                         result => result.TaskId);
```

Comment:
Are we sure each task has one, and only one, result ?

Reply:
By construction, yes

Reply:
yep only one result for one task. Multi results aren't yet implemented